byoqm/writer/writer.py

```diff
@@ -0,0 +1,116 @@
+import pandas as pd
+import logging
+import time
+import csv
+
+from typing import Dict
+from pathlib import Path
+
+from byoqm.metric.result import Result
+
+
+class Writer:
+    def __init__(
+        self,
+        model_name,
+        src_root,
+        output_dir,
+    ):
+        self._model_name = model_name
+        self._shortened_path = src_root
+        self._output_dir = output_dir
+        current_time: str = time.strftime("%Y-%m-%d_%H-%M-%S", time.gmtime())
+        self._file_name = Path(current_time + ".csv")
+
+    def run(self, results):
+        """
+        run ensures that the output path exists and thereafter starts saving the
+        respective files
+        """
+        if results == None:
+            logging.error("Results is none. Skipping writing...")
+            return
+        self._gen_output_paths_if_not_exists()
+        self._write_to_csv(results)
+
+    def _write_to_csv(self, results: Dict):
```

Comment:
At this point we should name it as if it is a public method i.e. drop `_` as a prefix.